src/Npgsql/NpgsqlParameter.cs

```diff
@@ -532,12 +532,21 @@ internal void ResolveTypeInfo(PgSerializerOptions options)
         var previouslyResolved = ReferenceEquals(typeInfo?.Options, options);
         if (!previouslyResolved)
         {
-            var dataTypeName =
-                _npgsqlDbType is { } npgsqlDbType
-                    ? npgsqlDbType.ToDataTypeName() ?? npgsqlDbType.ToUnqualifiedDataTypeNameOrThrow()
-                    : _dataTypeName is not null
-                        ? Internal.Postgres.DataTypeName.NormalizeName(_dataTypeName)
-                        : null;
+            string? dataTypeName;
+            if (options.DbTypeResolver.GetDataTypeName(DbType, options) is { } mappedDbTypeName)
```

Comment:
Could you make this branch short circuit when there are no dbtyperesolvers? That would keep this feature 'pay-for-play'.